### ISSUE TITLE
Unable to use logloss as score function

### ISSUE THREAD

**Issue by bear0330** [rhiever/tpot#238]
tpot 0.5 supports all sklearn score functions but if I use logloss as score function, I will get this error:
![image](https://cloud.githubusercontent.com/assets/9865603/17934076/26ed6130-6a49-11e6-9af4-2a62090b0838.png)

## Context of the issue

If I use logloss function like this:
```
tpot = TPOT(generations=5, scoring_function='log_loss', random_state=42, 
            verbosity=2)
```

I will get the error above. I think this is because the code inside tpot.py line 302~307:

```
               top_score = 0.
                for pipeline_num, pipeline in enumerate(self.hof.items):
                    if self.hof.keys[pipeline_num].wvalues[1] > top_score:
                        self._optimized_pipeline = pipeline
```

here the top_score is started with 0, but logloss will be negative value, that causes the condition won't hit then self._optimized_pipeline won't be assigned.

This is just my guess, I haven't test it yet, but if I don't use logloss, this error disappears.

## Process to reproduce the issue

1. User creates TPOT instance
2. User calls TPOT `fit()` function with training data and use logloss as score function.
3. TPOT crashes with a `TypeError`.

## Expected result

It should work with logloss function.

## Current result

Crash.

## Possible fix

See my guess above.

## `name of issue` screenshot

See introduction.

**Comment by rhiever** [comment]
Checking: Are you using TPOT for a regression problem? TPOT doesn't currently support regression, but we hope to do so in the near future (0.6).

Thank you for reporting this issue! This made me realize that we should place a limit on what scoring functions that TPOT accepts at the moment, as log loss is not appropriate for classification problems.

**Comment by rhiever** [comment]
Docs are updated with the latest push. Sorry for the confusion!

**Comment by bartleyn** [comment]
@rhiever There is a change of mine that got lost in the 0.5.0 release that accounts for the loss functions, which I think might be (at least) part of the reason for this. I agree that there should maybe be limits on what scoring functions TPOT uses, but I can put this back in if you like.

**Comment by rhiever** [comment]
Sounds good to me @bartleyn -- please roll it out as a patch to the dev branch.
